Expensify/App
# Bump react-native-web to 0.15.7 react to 17.0.2

### Details
cc @roryabraham @sketchydroide 

Coming from https://github.com/Expensify/Expensify.cash/pull/2939

I'm gonna recommend that we do not update to react-native-web 0.16 yet because there is a change to `EventEmitter` that is causing logs to stream in the console. Probably this will change when we update other dependencies that reference the more recent versions of `react-native`.

### Fixed Issues (Snyk PR)
https://github.com/Expensify/Expensify.cash/pull/2939

### Tests
### QA Steps
Test opening links and verify they open in new tabs on web
Test for general regressions in all platforms

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
